### PR TITLE
Fix docs after choice form type refactor

### PR DESCRIPTION
**Pull request by aivus**
Changes related with http://symfony.com/blog/new-in-symfony-2-7-choice-form-type-refactorization

### CODE REVIEW

File: reference/forms/types/entity.rst [hunk 1]
```diff
@@ -110,11 +110,12 @@ choice_label
     The ``choice_label`` option was introduced in Symfony 2.7. Prior to Symfony
     2.7, it was called ``property`` (which has the same functionality).
 
-**type**: ``string``
+**type**: ``string`` or a Closure
```
Comment (xabbuh) [inline reference/forms/types/entity.rst]: I would rather write `string|Closure` instead.

Comment (Tobion) [inline reference/forms/types/entity.rst]: actually it should be `null|callable|string|PropertyPath` see https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Form/ChoiceList/Factory/PropertyAccessDecorator.php#L162

**Comment by xabbuh** [comment]
You can now also pass callables to the `choice_name`, `choice_value`, `choice_attr` and `preferred_choices` options. Can you please also describe the arguments that are passed to the callables for each option?

Furthermore, you can also pass any `Traversable` object, a property path or `null` to these options.

I think we should also add a `versionadded` directive to each of these options which explains what has changed in Symfony 2.8 (this could be the same text for all options).

@aivus I know this seems like a lot of work, but I think it can be done for one option and then duplicated for the others accordingly. Do you think you can make these changes?

File: reference/forms/types/entity.rst (hunk 1 above)
Comment (xabbuh) [inline reference/forms/types/entity.rst]: exactly (see my comment [below](https://github.com/symfony/symfony-docs/pull/5755#issuecomment-146308077))

**Comment by weaverryan** [comment]
I believe this is also related to #5847. I mention that because we need to make these updates all at once, to make sure we've covered everything and done it correctly.

**Comment by weaverryan** [comment]
Thanks @aivus! I've included your commits in #5876, which will be merge soon.

**Pull request closed without merging**